src/main/java/mate/academy/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,67 @@
+package mate.academy.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.cartitem.CartItemDto;
+import mate.academy.dto.cartitem.CreateCartItemDto;
+import mate.academy.dto.cartitem.QuantityCartItemDto;
+import mate.academy.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.service.ShoppingCartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for mapping shopping cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Get shopping cart",
+            description = "Get all information of  user shopping cart")
+    public ShoppingCartDto getShoppingCart(Authentication authentication, Pageable pageable) {
+        return shoppingCartService.getShoppingCart(authentication.getName(), pageable);
```

Comment:
you can access user in next way:
```suggestion
        User user = (User) authentication.getPrincipal();
        return shoppingCartService.getShoppingCart(user.getId(), pageable);
```